### DISCUSSION TITLE
Edit Anchor Boxes sizes in model config

### DISCUSSION THREAD

**Discussion by FraPochetti** [rwightman/efficientdet-pytorch#220] [Q&A] [answered]
Hi all,
I see [here](https://github.com/rwightman/efficientdet-pytorch/blob/301487e859fa8160dd3e01b7dbc54d713b392676/effdet/config/model_config.py#L34) that I can change the anchors aspect ratios in the model config.
What about the anchors sizes?
E.g. this is what I would do in `torchvision` 
```python
AnchorGenerator(sizes=(32, 64, 128, 256, 512,), aspect_ratios=(0.25, 0.5, 1.0, 2.0, 4.0))
```
Is this supported in EffDet?
Or maybe the `sizes` are being automatically/dynamically inferred based on image size (and/or other params)?

Thanks!

**Comment by rwightman** [comment, marked as answer]
The box sizes are computed relative to the feature map sizes w/ the aspect ratios and anchor scale config. I based that on code from the TF impl, which shares most of the impl with Tensorflow Retinanet models, although some variations of that code do alow explicitly setting the sizes... 

It's a non-trivial amount of code to wrap ones head around, and have a TODO to simplify and maybe switch to the torchvision approach but it's well off of current priorities and projects....

I think you can tune it fairly easily for datasets where the scales of the objects differ from a typical coco by tuning the anchor scale (which can be specified as one scalar or a sequence per feature level.
https://github.com/rwightman/efficientdet-pytorch/blob/b3dcb20e45a4d5e3ad35146a9723de2d41aae999/effdet/anchors.py#L235-L285

**Reply by FraPochetti** [reply]
Thanks Ross!
